sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -1711,6 +1708,17 @@ class CastSuite extends CastSuiteBase {
     }
   }
 
+  test("SPARK-35698: cast timestamp without time zone to string") {
+    Seq(
+      "0001-01-01 00:00:00", // the fist timestamp of Common Era
+      "1582-10-15 23:59:59", // the cutover date from Julian to Gregorian calendar
+      "1970-01-01 00:00:00", // the epoch timestamp
+      "9999-12-31 23:59:59"  // the last supported timestamp according to SQL standard
+    ).foreach { s =>
+      checkEvaluation(cast(LocalDateTime.parse(s.replace(" ", "T")), StringType), s)
```

Comment:
can we put `T` in the strings directly?

Reply:
well, the result won't contain the "T", so we have to replace one of them eventually.
The test inputs are from https://github.com/apache/spark/blob/master/sql/catalyst/src/test/scala/org/apache/spark/sql/RandomDataGenerator.scala#L156

Reply:
ah I see, got it